uvtools/dspec.py

```diff
@@ -929,45 +948,50 @@ def dayenu_filter(x, data, wgts, filter_dimensions, filter_centers, filter_half_
         #if the axis orthogonal to the iteration axis is to be filtered, then
         #filter it!.
         for sample_num, sample, wght in zip(range(data.shape[fs-1]), _d, _w):
-            filter_key = (data.shape[fs], ) + tuple(np.round(x[fs],hash_decimal)) + tuple(filter_centers[fs]) + \
-            tuple(filter_half_widths[fs]) + tuple(filter_factors[fs]) + tuple(wght.tolist()) + ('inverse',)
+            filter_key = (data.shape[fs], ) + tuple(np.round(x[fs],hash_decimal))\
+             + tuple(np.round(np.asarray(filter_centers[fs]) * np.mean(np.diff(x[fs])) * len(x[fs]), hash_decimal))\
```

Comment:
seems like you should write a `_hash_array(arr, hash_decimal)` method that you should then call here. Makes this much easier to read.

Reply:
I've replaced these filter key generation lines with a `_hash_fourier_filter` method.